AtDigitization/AtPulse.h

```diff
@@ -41,6 +41,8 @@ class AtPulse {
    int fTBEntrance = 0;       //! Window location in timebuckets (from config)
    int fTBPadPlane = 0;       //! Pad plane location in TBs (calculated from DriftVelocity, TBEntrance, ZPadPlane
 
+   double fScaling = 1; // Scales the trace by this amount;
+
```

Comment:
What is the point of this? Doesn't it just scale the gain? Wouldn't it be simpler to adjust the gain parameter?

Reply:
The gain is taken from the parameter file. An adjustable scaling factor is required because our measured response function isn't perfectly normalized. It could be done with gain, but would require either exposing the gain to be a user defined value or making a different parameter file depending on the used response function, neither of which seem appropriate.

Reply:
If it's to compensate for differences in the response function, that feels like it should be folded into that rather than a misc other parameter in AtPulseTask. I'd prefer a response-function dependent gain parameter over another random scaling factor.

Reply:
Where might that go?

Reply:
What response function is the problem?

Reply:
Could add it to the base class https://github.com/ATTPC/ATTPCROOTv2/blob/develop/AtTools/ElectronicResponse/AtElectronicResponse.h.

If could be enforced by making the virtual functions protected and have non-virtual public classes that return the result of the protected * scalingFactor.

Reply:
Actually the class is already setup that way so this:
```
class AtElectronicResponse {
protected:
   virtual double GetResponse(double time) const = 0;
   virtual double GetResponse(int padNum, double time) const = 0;

public:
   virtual ~AtElectronicResponse() = default;
   /**
    * Returns the response of the electronics to a delta-function of charge at t=0 after time (us).
    */
   double operator()(double time) const { return GetResponse(time); }
   /**
    * Returns the response of the electronics to a delta-function of charge at t=0 after time (us).
    */
   double operator()(int padNum, double time) const { return GetResponse(padNum, time); }
};
```

Becomes

```
class AtElectronicResponse {
private:
double  fScale{1};
protected:
   virtual double GetResponse(double time) const = 0;
   virtual double GetResponse(int padNum, double time) const = 0;

public:
   virtual ~AtElectronicResponse() = default;
   /**
    * Returns the response of the electronics to a delta-function of charge at t=0 after time (us).
    */
   double operator()(double time) const { return GetResponse(time) * fScale; }
   /**
    * Returns the response of the electronics to a delta-function of charge at t=0 after time (us).
    */
   double operator()(int padNum, double time) const { return GetResponse(padNum, time) * fScale; }
void SetScale(double scale) { fScale = scale;}

};

Reply:
Then issue sorted without adding more random stuff to AtPulseTask.

Reply:
Honestly, I don't think a lot of the response function stuff existed when I was working on this. I was working out of a root file.